plugins/home/src/homePageComponents/Banner/Banner.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import Snackbar from '@material-ui/core/Snackbar';
+import classNames from 'classnames';
+import { BackstageTheme } from '@backstage/theme';
+import { makeStyles } from '@material-ui/core/styles';
+import SnackbarContent from '@material-ui/core/SnackbarContent';
+import { DateTime } from 'luxon';
+
+type BannerProps = {
+  startTime: string;
+  endTime: string;
+  info: React.ReactNode;
+  action?: React.ReactNode;
+  image?: string;
+  isOpen?: boolean;
+};
+const useStyles = makeStyles<
+  BackstageTheme,
+  { backgroundImage: string | undefined }
+>(
+  (theme: BackstageTheme) => ({
+    root: {
+      padding: theme.spacing(0),
+      marginBottom: theme.spacing(0),
+      marginTop: theme.spacing(0),
+      display: 'flex',
+      flexFlow: 'row nowrap',
+    },
+    topPosition: {
+      position: 'relative',
+      marginBottom: theme.spacing(6),
+      marginTop: -theme.spacing(3),
+      zIndex: 'unset',
+    },
+    icon: {
+      fontSize: 20,
+    },
+    content: {
+      width: '100%',
+      maxWidth: 'inherit',
+      flexWrap: 'nowrap',
+    },
+    message: {
+      display: 'flex',
+      alignItems: 'center',
+      color: theme.palette.text.primary,
+      fontWeight: Number(theme.typography.fontWeightBold),
+    },
+    info: {
+      backgroundColor: '#18a888',
+      backgroundImage: ({ backgroundImage }) => `url(${backgroundImage})`,
+    },
+  }),
+  { name: 'BackstageHomepageBanner' },
+);
+
+export const Banner = (props: BannerProps) => {
+  const classes = useStyles({
+    backgroundImage: props.image,
+  });
+  const shouldRenderByDate =
+    DateTime.fromISO(props.startTime) < DateTime.now() &&
+    DateTime.now() < DateTime.fromISO(props.endTime);
+
+  const isOpen =
+    typeof props.isOpen === 'undefined'
+      ? shouldRenderByDate
+      : props.isOpen && shouldRenderByDate;
```

Comment:
This should handle the logic to not render the banner when explicitly they say it shouldn't be rendered. But when it gets the explicit isOpen=true it still should check if it is the right time to render it. 
I added this to be able to pass in an action that should handle the close of the banner